### PR TITLE
fix(@ui-components/master): force TypeScript@~3.6.4

### PR DESCRIPTION
**Pull request by ericclemmons**
https://github.com/aws-amplify/amplify-js-samples-staging/pull/42#issuecomment-579427196 was blocked by TypeScript issues.

Investigation found that `^3.5.3` matches `3.7`, which is what has been installed for this branch for sometime.

This is effectively cherry-picking Ivan's 6597db2b0704d7aff5b612557536142b82e1731 that hasn't made it in since 788d7df7b2f1a3c02a6e6909e934de5fdc0b5954.

> https://github.com/aws-amplify/amplify-js/commits/ui-components/master/package.json

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -21649,10 +21649,10 @@ typescript@^3.3.4000:
   resolved "https://registry.yarnpkg.com/typescript/-/typescript-3.7.4.tgz#1743a5ec5fef6a1fa9f3e4708e33c81c73876c19"
   integrity sha512-A25xv5XCtarLwXpcDNZzCGvW2D1S3/bACratYBx2sax8PefsFhlYmkQicKHvpYflFS8if4zne5zT5kpJ7pzuvw==
 
-typescript@^3.5.3:
-  version "3.7.3"
-  resolved "https://registry.yarnpkg.com/typescript/-/typescript-3.7.3.tgz#b36840668a16458a7025b9eabfad11b66ab85c69"
```
Comment (ericclemmons) [inline yarn.lock]: Notice the resolved version is `3.7.3`.

File: packages/api-graphql/src/GraphQLAPI.ts [hunk 2]
```diff
@@ -243,7 +243,7 @@ export class GraphQLAPIClass {
 			},
 		};
 
-		const endpoint = customGraphqlEndpoint ?? appSyncGraphqlEndpoint;
+		const endpoint = customGraphqlEndpoint || appSyncGraphqlEndpoint;
```
Comment (jordanranz) [inline packages/api-graphql/src/GraphQLAPI.ts]: Is this related to the typescript change?

Comment (ericclemmons) [reply packages/api-graphql/src/GraphQLAPI.ts]: Yep: https://github.com/aws-amplify/amplify-js/pull/4452
